src/builds/appveyor.rs

```diff
@@ -0,0 +1,88 @@
+use std::io::Read;
+
+use chrono::{DateTime, UTC};
+use diesel;
+use diesel::prelude::*;
+use hyper::Client;
+use hyper::header::Accept;
+use hyper::net::HttpsConnector;
+use hyper_native_tls::NativeTlsClient;
+use serde::de::DeserializeOwned;
+use serde_json;
+
+use DB_POOL;
+use domain::builds::Build;
+use error::DashResult;
+
+static ROOT_URL: &'static str = "http://ci.appveyor.com/api";
+
+pub fn get_and_insert_build(build: &str) -> DashResult<()> {
+    let url = format!("{}/projects/rust-lang/rust/build/{}", ROOT_URL, build);
+    let response: ResponseFromJson = get(&url)?;
+
+    let conn = &*DB_POOL.get()?;
+    for job in response.build.jobs.iter() {
+        if job.finished.is_none() || job.status == "cancelled" {
+            continue;
+        }
+
+        let duration = job.finished.unwrap().signed_duration_since(job.started);
```

Comment:
It'd be nice to make the unwrap unnecessary so the control flow is clearer, maybe?

Something like this? It needs two continues which isn't great but it'd reduce the likelihood of accidentally introducing panics later on.

```rust
let duration = match job.finished {
    Some(f) => {
        if job.status == "cancelled" { continue; }
        f.signed_duration_since(job.started)
    }
    None => continue,
}
```